### ISSUE TITLE
upgrade from 1.3.1 to 2.0.0 breaks app

### ISSUE THREAD

**Issue by aoreinoso** [carrierwaveuploader/carrierwave#2413]
Gemfile
========
	gem 'carrierwave'
	gem 'mini_magick'
	gem 'mongoid'
	gem 'bson_ext'
	gem 'carrierwave-mongoid'
	
uploader: image_uploader.rb
=========
	include CarrierWave::MiniMagick
	#
	storage :grid_fs
	#
	def store_dir
		"#{model.class.to_s.underscore}/#{mounted_as}/#{model.id}"
	end
	#
	def rotate(strGrados)
		manipulate! do |img|
			img.rotate strGrados
			img
		end
	end

model: imagen.rb
======
	mount_uploader :imagen, ImageUploader

controller: imagens_controller.rb
===========
	def rotate
                @imagen = Imagen.find(params[:id])
		begin
			...
			...
			@imagen.imagen.rotate(90)
			...
		rescue => e
			logger.info("Error rotando la imagen: #{e.inspect}")
		end
	end

===========================
carrierwave 1.3 all is ok

carrierwave 2.0
#<Errno::ENOENT: No such file or directory @ rb_sysopen - uploads/tmp/1566544774-44891208113911-0001-6290/mini_magick20180711-30587-fjzb8i.jpg>

Thanks in advance

**Comment by shanecav84** [comment]
Looks like https://github.com/carrierwaveuploader/carrierwave/commit/629afecbaeccd2300e4660b78ee36bd95dd845c5 changed the default `cache_storage` from `:file` to `nil`. Setting `config.cache_storage = :file` fixes the issue for me.

**Comment by aoreinoso** [comment]
config.cache_storage = :file It works for me but only once
I get the same result when I try to rotate the same image a second time

**Comment by mshibuya** [comment]
Could you paste full stack trace? Just the message is not enough...

**Comment by aoreinoso** [comment]
In my example When I see the image the system loads:
"filename"=>"imagen/imagen/11216/image_mini_magick20180711-30587-1c34uqi.jpg"
When I enter it the system loads:
"filename"=>"imagen/imagen/11216/mini_magick20180711-30587-1c34uqi.jpg"
Now when the rotate gives error I have to image:
"filename"=>"imagen/imagen/11216/mini_magick20180711-30587-1c34uqi.jpg"
It is rotated but not the image:
"filename"=>"imagen/imagen/11216/image_mini_magick20180711-30587-1c34uqi.jpg"
I guess that's where the error:
#<Errno::ENOENT: No such file or directory - getcwd>

Sorry if I do not explain myself well because it is not easy I have been doing tests for days and I cannot find the error although I am sure it must have something to do with the cache. After the error I have to restart nginx to try it again

**Comment by aoreinoso** [comment]
Sorry,
Everything is fine, in image_uploader I was also processing other data and rotating the file was changing it then I could not access it.
The solution has been quite complex, the funny thing is that in 1.3 this did not happen

**Comment by coding-bunny** [comment]
Upgrading the gem in our case results in all md5 checksums not being populated in our records when duplicating images. All these tests fail in our suite when we upgrade from 1.3.1 to 2.0.1

**Comment by JanStevens** [comment]
Could this be added to the upgrading / deprecation list? Or maybe a general gotcha upgrade notice for 1 -> 2?

Or maybe the default should be `file` again instead of `nil`?

**Comment by mshibuya** [comment]
@coding-bunny Yours seems not related to this one, please consider opening a new issue.

@JanStevens I need to take a look at details like full stack trace and application code to investigate if this is a real issue. Can you provide one first?

**Comment by coding-bunny** [comment]
@mshibuya : Setting `config.cache_storage = :file` resolves the upload problems for us at least as well, will try later to see if this also resolves the md5 problems.
